test/WebJobs.Script.Tests/ScriptHostManagerTests.cs

```diff
@@ -59,6 +61,34 @@ public class ScriptHostManagerTests
             }
         }
 
+        [Fact]
+        public void RunAndBlock_DisposesOfHost_WhenExceptionIsThrown()
```

Comment:
Thanks for adding test. We need to get in the habit of this for all new code now, to stop the debt :)